### DISCUSSION TITLE
Assistance with Custom Lock File Path in Monorepo Configuration

### DISCUSSION THREAD

**Discussion by AnaGabrieleSoares** [renovatebot/renovate#30728] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab - renovate:37.427.0

### Please tell us more about your question or problem

Hi Renovate Team,  
   
My team is looking to implement Renovate in a monorepo to help manage dependencies. Our lock file, `pnpm-lock.yml`, is not located in the root directory. Instead, they are in `/common/config/rush`. I don’t see a way in the Renovate docs to configure a custom lock file path.  
   
Could you please point out the best way to specify a custom path for the lock file (`pnpm-lock.yml`) in the Renovate configuration? Additionally, are there any best practices for handling lock files that are not in the root directory, especially in the context of a monorepo?  
   
Thank you for your assistance!  
  

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
Out of curiosity! Why do you have the lockfile in a separate folder instead of the root?

**Reply by opnsrce** [reply]
The lockfile is in a separate folder because the repo is a mono repo built using Rush. Rush creates an interface over the lock file and manages it.

**Comment by viceice** [comment]
- https://github.com/renovatebot/renovate/issues/3681
